src/coreclr/vm/arm64/asmhelpers.S

```diff
@@ -653,7 +653,8 @@ DynamicHelper DynamicHelperFrameFlags_ObjectArg | DynamicHelperFrameFlags_Object
 
 LEAF_ENTRY JIT_GetDynamicNonGCStaticBase_SingleAppDomain, _TEXT
     // If class is not initialized, bail to C++ helper
-    ldr x1, [x0, #OFFSETOF__DynamicStaticsInfo__m_pNonGCStatics]
+    add x1, x0, #OFFSETOF__DynamicStaticsInfo__m_pNonGCStatics
+    ldar x1, [x1]
```

Comment:
Do we need the barriers in arm32 version of these helpers as well?

Reply:
(And also riscv and loongarch.)

Reply:
Yes we do, this was more intended to get a quick bit written up so that we could run some tests.

Reply:
NOTE: can be `ldapur x1, [x1, #OFFSETOF__DynamicStaticsInfo__m_pGCStatics]` for macos-arm64 since that has rcpc2 in the baseline, but not sure it worth the complexity.

Reply:
Also, it's a detail that we will be able to do a better job of implementing these helpers when we move them to managed. (Since the jit will take advantage of the ldapr or ldapur instructions instead of just using ldar which is a stronger barrier than we need.)

Reply:
Finally, there is no need to fix arm, as we don't have handwritten assembly helpers for this set of helpers.